napari/_vispy/_tests/test_utils.py

```diff
@@ -60,11 +67,52 @@ def test_get_view_direction_in_scene_coordinates_2d(make_napari_viewer):
     viewer.dims.ndisplay = 2
 
     # get the viewbox
-    view_box = viewer.window._qt_viewer.view
+    view_box = viewer.window._qt_viewer.canvas.view
 
     # get the view direction
     view_dir = get_view_direction_in_scene_coordinates(
         view_box, viewer.dims.ndim, viewer.dims.displayed
     )
 
     assert view_dir is None
+
+
+@pytest.mark.parametrize('cursor', cursors_param)
+def test_set_cursor(cursor, make_napari_viewer):
```

Comment:
I have a few issues with this test.

The main one is that it's expensive because it makes many calls to `make_napari_viewer`, but doesn't test much of substance and mostly just [reflects the implementation](https://github.com/melonora/napari/blob/667e8afb1008d10be1eedd35698096b58aaa389c/napari/_vispy/canvas.py#L248).

The other issue I have is that the nested if/else blocks make it hard to read and reason about. It would be far simpler to split each block into its own test like the following.

```python
@pytest.mark.parametrize('cursor', ('cross', 'forbidden', 'pointing', 'standard')
def test_set_cursor_visual(cursor, make_napari_viewer):
    viewer = make_napari_viewer()
    viewer.cursor.style = cursor
    assert viewer.window._qt_viewer.canvas.cursor.shape() == QtCursorVisual[cursor].value

def test_set_cursor_invalid(make_napari_viewer):
    viewer = make_napari_viewer()
    with pytest.raises(Exception):
        viewer.cursor.style = 'invalid'
...
```

My suggestion is either remove this test entirely, or just test the interesting paths - the adaptive behavior of the square and circle, and maybe the invalid value.

Reply:
Adjusted with [01a07bc](https://github.com/napari/napari/pull/5432/commits/01a07bc01b2cf9206ab9dc8d6a9a1765c3b24d26)